@shared/api/types.ts

```diff
@@ -22,10 +22,16 @@ export interface Response {
   type: SERVICE_TYPES;
   url: string;
   isDataSharingAllowed: boolean;
-  allAccounts: Array<string>;
+  allAccounts: Array<Account>;
 }
 
 export interface ExternalRequest {
   transactionXdr: string;
   type: EXTERNAL_SERVICE_TYPES;
 }
+
+export interface Account {
+  publicKey: string;
+  name: string;
+  imported: boolean;
+}
```

Comment:
this is the main thing to see in this PR: rather an array of publicKeys, allAccounts will now be an array of type Account so the FE has all the information about the account in a contained object